Expose core modules as `pub mod`

All modules in `lib.rs` are declared with `mod` (private visibility). While Python consumers access types through the `#[pymodule]` registration, Rust consumers who depend on the crate via `rlib` cannot access any of the engine's types or functions.

```rust
// lib.rs (current)
mod action;       // Action, ActionType, Phase
mod observation;  // Observation
mod parser;       // tid_to_mjai(), mjai_to_tid()
mod rule;         // GameRule
mod state;        // GameState
```

When adding this crate as a Rust dependency (the `Cargo.toml` already produces an `rlib`), imports fail:

```
error[E0603]: module `action` is private
 --> src/game.rs:4:17
  |
4 | use _riichienv::action::{Action, ActionType, Phase};
  |                 ^^^^^^ private module
```

## Proposed Change

Make modules required by external Rust consumers public:

```rust
pub mod action;         // Action, ActionType, Phase
pub mod observation;    // Observation
pub mod parser;         // tid_to_mjai(), mjai_to_tid()
pub mod rule;           // GameRule, KuikaeMode, KanDoraTimingMode
pub mod state;          // GameState, PlayerState, WallState
```

Note: `types` likely needs to be `pub mod` as well, since `Meld`, `MeldType`, and `WinResult` appear in the public API of `Observation` and `GameState`.

## Impact

- No change to the Python binding — `#[pymodule]` registration is independent of module visibility.
- The `state` sub-modules (`event_handler`, `legal_actions`, `player`, `wall`) are already `pub mod`.

--- a/native/src/lib.rs
+++ b/native/src/lib.rs
@@ -4,17 +4,17 @@ mod agari;
 mod hand_evaluator;
 mod score;
 mod tests;
-mod types;
+pub mod types;
 mod yaku;
 
-mod action;
+pub mod action;
 mod env;
-mod observation;
-mod parser;
+pub mod observation;
+pub mod parser;
 mod replay;
-mod rule;
+pub mod rule;
 mod shanten;
-mod state;
+pub mod state;
 pub mod win_projection;
 mod yaku_checker;
 

